polaris-core/src/main/java/org/apache/polaris/core/secrets/UnsafeInMemorySecretsManager.java

```diff
@@ -73,9 +73,9 @@ public UserSecretReference writeSecret(
 
     String secretUrn;
     for (int secretOrdinal = 0; ; ++secretOrdinal) {
+      String typeSpecificIdentifier = forEntity.getId() + ":" + secretOrdinal;
       secretUrn =
-          String.format(
-              "urn:polaris-secret:unsafe-in-memory:%d:%d", forEntity.getId(), secretOrdinal);
+          UserSecretReferenceUrnHelper.buildUrn(SECRET_MANAGER_TYPE, typeSpecificIdentifier);
```

Comment:
Can we not just add `buildUrn` or something like it to `UserSecretsManager`?

Reply:
Done.

Although I might have missed something here, please let me know if you think the buildUrn function meets your intended goal.

Reply:
Yeah it looks good, though seems like callers could just directly use `UserSecretReference.buildUrnString`. If there's a situation where the implementation might diverge, having the additional `buildUrn` method makes sense. My original comment was more about the necessity of `UserSecretReferenceUrnHelper`.

Reply:
Makes sense.